### PR TITLE
Item now supports IP based auth

### PR DESCRIPTION
**Pull request by grepsedawk**

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -4,5 +4,5 @@ class Item < ApplicationRecord
 
   has_and_belongs_to_many :auth_ip_addresses
   has_and_belongs_to_many :auth_services
-  belongs_to :end_service
+  belongs_to :end_service, optional: true
```
Comment (eshadatta) [inline app/models/item.rb]: Hi @pachonk ,

@ekate is looking at the whole PR. But I had a question about this line. Could you tell me why belongs_to end_service is optional here? Are you thinking there would be a use case where an item doesn't belong to an end service?  Thanks.

Comment (grepsedawk) [reply app/models/item.rb]: @eshadatta Thanks for the catch! Early on for this feature addition, I had confused end_service and auth_service, so I switched end_service to optional in order to allow for ip or auth_service based auth.
Reverted, Thanks!

**Comment by jgpawletko** [comment]
Hi @pachonk ,

This looks good to me. (I resolved my questions on my own ;-) ).

Cc: @ekate @eshadatta

**Comment by grepsedawk** [comment]
Thanks for reviewing @jgpawletko! After I get confirmation from @ekate and/or @eshadatta, I'll merge this and then rebase the other PR (https://github.com/NYULibraries/dlts-middleware-authz/pull/20) and reopen it.

**Comment by eshadatta** [comment]
Hi @pachonk ,

Yes, please go ahead and merge/rebase this. Thanks!

**Pull request merged**